### PR TITLE
Fix xz_utils on Mingw

### PR DESCRIPTION
**Pull request by uilianries**
Specify library name and version:  **xz_utils/5.2.4**

fixes https://github.com/conan-io/conan-center-index/issues/750

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/xz_utils/all/conanfile.py [hunk 1]
```diff
@@ -25,7 +25,7 @@ def _is_mingw_windows(self):
         return self.settings.compiler == 'gcc' and self.settings.os == 'Windows' and os.name == 'nt'
 
     def build_requirements(self):
-        if self._is_mingw_windows:
+        if self._is_mingw_windows and "CONAN_BASH_PATH" not in os.environ:
```
Comment (SSE4) [inline recipes/xz_utils/all/conanfile.py]: we also sometimes use more complicated condition:
 `"CONAN_BASH_PATH" not in os.environ and tools.os_info.detect_windows_subsystem() `
do you think if it's more reliable? in which cases it is needed?

Comment (danimtb) [reply recipes/xz_utils/all/conanfile.py]: @uilianries ping

Comment (uilianries) [reply recipes/xz_utils/all/conanfile.py]: @SSE4 is right, this case is more specific. Checking msys2 can avoid possible problems related to misconfiguration, like it's msys2 but is not detect as is.

Comment (uilianries) [inline recipes/xz_utils/all/conanfile.py]:
```suggestion
        if self._is_mingw_windows and "CONAN_BASH_PATH" not in os.environ tools.os_info.detect_windows_subsystem() != 'msys2':
```

Comment (uilianries) [reply recipes/xz_utils/all/conanfile.py]: Done! I've added @SSE suggestion. Thanks for reviewing!

**Pull request merged**